src/Microsoft.Extensions.Diagnostics.HealthChecks/DefaultHealthCheckService.cs

```diff
@@ -77,7 +77,9 @@ internal class DefaultHealthCheckService : HealthCheckService
 
                             Log.HealthCheckEnd(_logger, registration, entry, stopwatch.GetElapsedTime());
                         }
-                        catch (Exception ex)
+
+                        // Allow cancellation to propagate.
+                        catch (Exception ex) when (ex as OperationCanceledException == null)
```

Comment:
Is this easier to grock: `when (!(ex is OperationCanceledException))`

Reply:
eh, I like it the way it is